src/components/Api.js

```diff
@@ -1,45 +1,98 @@
-class Api {
+export default class Api {
   constructor(options) {
-    this._authorization = "590f97d9-e5fa-4775-8c94-b9976acf5893";
+    this._headers = options.headers;
+    this._baseUrl = options.baseUrl;
+  }
+  // Promise
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error ${res.status}`);
+    }
+  }
+
+  _request(url, options) {
+    return fetch(url, options).then(this._handleResponse);
   }
 
+  // Cards
+
   getInitialCards() {
-    return fetch("https://around-api.en.tripleten-services.com/v1", {
-      headers: {
-        authorization: this._authorization,
-      },
-    })
-      .then((res) => {
-        if (res.ok) {
-          return res.json();
-        }
-        // if the server returns an error, reject the promise
-        return Promise.reject(`Error: ${res.status}`);
-      })
-      .catch((err) => console.error(err));
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._handleResponse);
```

Comment:
**Could be improved**

Please, use `_request` everywhere not to duplicate `.then(this._handleResponse);`

```js

  getInitialCards() {
    return this._request(`${this._baseUrl}/cards`, {
      headers: this._headers,
    })
  }
```